packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -48,6 +57,69 @@ export type CustomSelectProps = {
 	 */
 	value?: string | string[];
 };
+/**
+ * The legacy object structure for the options array.
+ */
+type Option = {
+	key: string;
+	name: string;
+	style?: {};
+	className?: string;
+	__experimentalHint?: string;
+};
+
+/**
+ * The legacy object returned from the onChange event.
+ */
+type OnChangeObject = {
+	selectedItem: Option;
+	highlightedIndex?: number;
+	isOpen?: boolean;
+};
+
+export type LegacyCustomSelectProps = {
+	/**
+	 * Used to visually hide the label. It will always be visible to screen readers.
+	 *
+	 */
+	hideLabelFromVision?: boolean;
+	/**
+	 * Pass in a description that will be shown to screen readers associated with the
+	 * select trigger button. If no value is passed, the text "Currently selected:
+	 * selectedItem.name" will be used fully translated.
+	 */
+	describedBy?: string;
+	/**
+	 * Label for the control.
+	 */
+	label: string;
+	/**
+	 * A function that receives the new value of the input.
+	 */
+	onChange?: ( newValue: OnChangeObject ) => void;
+	/**
+	 * The options that can be chosen from.
+	 */
+	options: Array< Option >;
+	/**
+	 * The size of the control.
+	 *
+	 * @default 'default'
+	 */
+	size?: CustomSelectProps[ 'size' ];
```

Comment:
It looks like the legacy component [supports the same values for `size` as `SelectControl`](https://github.com/WordPress/gutenberg/blob/d2450c88f42e22827718ca233a94cb9c5e1b873f/packages/components/src/custom-select-control/index.js#L78-L79), which are ultimately the same as `InputControl`.

Question is: are all those different values effectively having an impact on the legacy `CustomSelectControl`? If so, we should probably accept all those values and retain the correct behavior.

Otherwise, we could be able to trim down the list of accepted `size` values.